src/test/java/tutorlink/command/ListComponentCommandTest.java

```diff
@@ -0,0 +1,53 @@
+package tutorlink.command;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+
+import java.util.HashMap;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import tutorlink.appstate.AppState;
+import tutorlink.component.Assignment;
+import tutorlink.component.Exam;
+import tutorlink.exceptions.TutorLinkException;
+import tutorlink.result.CommandResult;
+
+public class ListComponentCommandTest {
+
+    private AppState appState;
+    private ListComponentCommand command;
+
+    @BeforeEach
+    public void setUp() {
+        appState = new AppState();
+        command = new ListComponentCommand();
+    }
+
+    @Test
+    void execute_emptyList_showsEmptyMessage() throws TutorLinkException {
+        CommandResult result = command.execute(appState, new HashMap<>());
+        assertEquals("No components found.", result.toString());
+    }
+
+    @Test
+    void execute_nonEmptyList_showsAllComponents() throws TutorLinkException {
+        appState.components.addComponent(new Assignment("Assignment 1", 50.0, 0.3));
+        appState.components.addComponent(new Exam("Midterm", 100.0, 0.4));
+
+        CommandResult result = command.execute(appState, new HashMap<>());
+
```

Comment:
Line 55-60 of ComponentList.java will lead to formatting issues where the first index will be misaligned with the rest of the index.

Consider changing follow line 55-60 of StudenList.java to align the indices

This might cause changes in the test cases, which will need to be updated. Else, LGTM